core/docker/bin/run-trino

```diff
@@ -7,4 +7,6 @@ if ! grep -s -q 'node.id' /etc/trino/node.properties; then
     launcher_opts+=("-Dnode.id=${HOSTNAME}")
 fi
 
+launcher_opts+=("-J--add-modules=jdk.incubator.vector")
```

Comment:
I think we should add this to jvm.config instead

Reply:
We could, but it's easy to miss and it'll fail queries at runtime only when parquet is used while the server will start-up without complaining.

Reply:
As with the other recommended/required JVM settings. We can detect whether when enabled the module is linked. Please see `verifyPackageAccessAllowed` in `BigQueryConnectorModule`. We could do the same when vectorized parquet reader is enabled

Reply:
That approach would also work, but it feels a bit roundabout to throw error about a required JVM property and make the user fill it in when we can easily just include it ourselves. But I don't mind going that way depending on what the consensus is.

Reply:
I don't think that modifying a launcher is a proper way of adding required JVM flags.

Reply:
I like the idea of automatically adding required JVM flags, but this should be done in the Airlift launcher, so that they're present no matter how Trino is run. The change here only affects Docker.

Reply:
Created https://github.com/airlift/airlift/pull/1144

Reply:
I have refactored this PR so that the Vector module is no longer required. We will automatically fall back to existing code paths if Vector module isn't found. So now it's just a matter of finding a convenient way to add `--add-modules=jdk.incubator.vector`. We could just add it to all the JVM configs in the codebase and add it to our JVM configs documentation and leave it at that. The other option is to add it to `core/docker/bin/run-trino` which will cover our main mode of deployment.